### PR TITLE
Concrete ReLU Solver

### PR DESCRIPTION
**Pull request by SebastianGuadalupe**

### CODE REVIEW

**Comment by mforets** [comment]
looks good so far, i think that we can merge this after resolving the merge conflict.

the evaluation of the output set with the lazy intersections is an orthogonal issue. what happens if you just override LazySets.rho(d, X \cap Y) using the upper bound rule? see also the ideas [here](https://github.com/JuliaReach/LazySets.jl/issues/175#issuecomment-583594153)

File: src/solve.jl
Comment (schillic) [inline src/solve.jl:133]: I do not understand this part. Above `U₀` is a set and here it is a vector. This will crash in one of those cases.

**Review by schillic — APPROVED** [review]

File: src/solve.jl
Comment (SebastianGuadalupe) [reply src/solve.jl]: You're right, even if it didn't crash, the code after it would.

**Pull request merged**
